application/helpers/frontend_helper.php

```diff
@@ -1865,6 +1865,10 @@ function doAssessment($surveyid, $returndataonly=false)
             $groups=array();
             foreach($fieldmap as $field)
             {
+                // Init Assessment Value
+                $assessmentValue = NULL;
```

Comment:
Yes : it's an issue of old code, but old code didn't send an event + didn't search in DB for all question type.